### DISCUSSION TITLE
Gotcha with subscribeWithSelector()

### DISCUSSION THREAD

**Discussion by dcorb** [pmndrs/zustand#1846] [Q&A] [answered]
selectedDate or selectedFacility is not changing, so refreshTimeline is not triggered (as expected).

```
const unsub = useSettingsStore.subscribe(
  (state) => state.selectedDate,
  () => {
    refreshTimeline();
  }
);

const unsub2 = useSettingsStore.subscribe(
  (state) => state.selectedFacility,
  () => {
    refreshTimeline();
  }
);
```

but here under the same conditions, refreshTimeline gets triggered (I didn't expect this).

```
const unsub3 = useSettingsStore.subscribe(
  (state) => [state.selectedFacility, state.selectedDate],
  () => {
    refreshTimeline();
  },
);
```

and here refreshTimeline doesn't get triggered. (So this is the equivalent of the 1st example?)

```

const unsub3 = useSettingsStore.subscribe(
  (state) => [state.selectedFacility, state.selectedDate],
  () => {
    refreshTimeline();
  },
  { equalityFn: shallow }
);
```

I'm missing a bit better documentation on this topic. I still don't understand why these difference in the behaviour.

**Comment by dai-shi** [comment, marked as answer]
This is about with `subscribeWithSelector` middleware, right?
`(state) => [state.selectedFacility, state.selectedDate],` creates a new array every time, so the behavior is understandable.

**Reply by dcorb** [reply]
yes, subscribeWithSelector.
I assumed (wrongly) my 1st and 2nd example would be equivalent. Just a more concise syntax

**Reply by dcorb** [reply]
Oh, I have the same wrong assumption here:

```
  const [a, b, c] = useOrderStore((state) => [
    state.a,
    state.b,
    state.c,
  ]);
```

This also triggers re-renders even if a, b or c didn't change (some other changes happening in the store).

It's ok, once you know this, it has sense now
